test_security/CMakeLists.txt

```diff
@@ -312,24 +316,44 @@ if(BUILD_TESTING)
     find_program(PROGRAM ros2)
 
     set(node_names_list "/publisher;/subscriber;/publisher_missing_key;/publisher_invalid_cert")
-    file(REMOVE_RECURSE "${KEYSTORE_DIRECTORY}/enclaves/publisher_invalid_cert")
-    set(generate_artifacts_command ${PROGRAM} security generate_artifacts -k ${KEYSTORE_DIRECTORY_NATIVE_PATH} -e ${node_names_list})
-    execute_process(
-      COMMAND ${generate_artifacts_command}
-      RESULT_VARIABLE GENERATE_ARTIFACTS_RESULT
-      ERROR_VARIABLE GENERATE_ARTIFACTS_ERROR
+
+    ament_add_test(pre_clean_artifacts
```

Comment:
These really don't need to be ament tests, but I made them such. I'm on the fence about converting them to plain ctest tests instead, especially since they can't really be 'selected' and are just run when needed.

Reply:
Yeah, from what I understand about fixtures (which isn't too much), these will always be run when they are needed, regardless of what you select.  From that perspective, I think it would make sense to make these regular ctest tests, since it is something of an indication that there is something special/different about them.

I think we could also use a comment in this section explaining why we are doing this here.

Reply:
Thanks for the review.

Alright, I converted to simple `add_test` calls and added some comments.

IMO, going into detail about why we're doing this during the test phase rather than configure phase isn't valuable. In my experience the existing approach was more of a deviation from the norm, and this approach is closer to what I've seen elsewhere in the ROS 2 codebase. That said, I did add more comments discussing what's being defined here.